pkg/rpcclient/rolemgmt/roles.go

```diff
@@ -98,3 +103,47 @@ func (c *Contract) DesignateAsRoleTransaction(role noderoles.Role, pubs keys.Pub
 func (c *Contract) DesignateAsRoleUnsigned(role noderoles.Role, pubs keys.PublicKeys) (*transaction.Transaction, error) {
 	return c.actor.MakeUnsignedCall(Hash, designateMethod, nil, int(role), pubs)
 }
+
+// FromStackItem converts provided [stackitem.Array] to DesignationEvent or returns an
+// error if it's not possible to do to so.
+func (e *DesignationEvent) FromStackItem(item *stackitem.Array) error {
+	if item == nil {
+		return errors.New("nil item")
+	}
+
+	arr, ok := item.Value().([]stackitem.Item)
+	if !ok {
+		return errors.New("not an array")
+	}
+	if len(arr) != 2 {
+		return errors.New("wrong number of event parameters")
+	}
+
+	roleBigInt, err := arr[0].TryInteger()
+	if err != nil {
+		return fmt.Errorf("invalid Role: %w", err)
+	}
+	if !roleBigInt.IsUint64() {
+		return fmt.Errorf("role overflow: %s", roleBigInt.String())
```

Comment:
There's an inconsistency in `Role` naming in errors here and below: it's either `Role` or `role` everywhere.